library/psa_crypto.c

```diff
@@ -3211,6 +3212,9 @@ psa_status_t psa_asymmetric_decrypt( mbedtls_svc_key_id_t key,
 
     *output_length = 0;
 
+    if( ! PSA_ALG_IS_ASYMMETRIC_ENCRYPTION( alg ) )
```

Comment:
I am not completely sure because the line numbers are not those of the last commit but it seems that the previous one is in the `psa_asymmetric_encrypt` API thus that the check is performed once at the beginning of each API which seems ok.